### DISCUSSION TITLE
Form submission rewrites the URL

### DISCUSSION THREAD

**Discussion by aloukissas** [remix-run/remix#1292] [Q&A] [answered]
I have a simple form, which after submissions seems to rewrite the URL (e.g. `http://localhost:3000` becomes `http://localhost:3000/?index`). Things work correctly but it's more of an aesthetics thing. I think it may have something to do with how in plain reactjs you need to do `e.preventDefault()` in the `onSubmit` handler.

**Comment by sergiodxa** [comment, marked as answer]
That’s because a form cause a navigation to the url of the action, that’s just how the forms work, you can redirect on the action to ensure the url is the original without the ?index at the end
